pkg/cache/cache.go

```diff
@@ -307,12 +323,17 @@ func (c *itemCache) set(thisAction cacheAction) (thisReply cacheReply) {
 	if _, ok := c.Items[key]; ok {
 		secKeys = c.Items[key].SecondaryKeys
 	}
+
 	c.Items[key] = &Item{
 		Object:        data,
 		UpdateTime:    time.Now().Unix(),
 		Hash:          hash,
 		SecondaryKeys: secKeys,
 	}
+
+	if data != nil && reflect.ValueOf(data).Type().Kind() == reflect.Ptr {
```

Comment:
You might need to also check if you have a pointer to nil. I added a func for it in the marketplace branch in util/util.go. It checks for nil, and a pointer that has a nil value. Turns out these are two different things.

```
// IsNil checks a value, or a pointer for nil
func IsNil(v interface{}) bool {
	return v == nil || reflect.ValueOf(v).Kind() == reflect.Ptr && reflect.ValueOf(v).IsNil()
}
```